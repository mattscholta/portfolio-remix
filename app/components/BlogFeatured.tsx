import { BlogPreview } from "~/components/BlogPreview";

export interface BlogFeaturedProps {
  className?: string;
}

export const BlogFeatured = (props: BlogFeaturedProps) => {
  const { className } = props;

  return (
    <div className={className}>
      <h2 className="mb-8 text-2xl">Featured Article</h2>
      <BlogPreview
        content={"item.content.html"}
        date={new Date().toISOString()}
        image={
          "https://mattscholta.files.wordpress.com/2017/08/pexels-photo-4164418.jpeg"
        }
        key={"item.id"}
        slug={"improving-code-quality"}
        title={"item.title"}
      />
    </div>
  );
};
