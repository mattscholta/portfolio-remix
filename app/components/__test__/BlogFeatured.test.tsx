import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import type { RenderResult } from "@testing-library/react";

import { BlogFeatured } from "../BlogFeatured";
import type { BlogFeaturedProps } from "../BlogFeatured";
import { MemoryRouter } from "react-router";

describe("BlogFeatured Component", () => {
  let props: BlogFeaturedProps;
  let component: RenderResult;

  const createWrapper = () => {
    component = render(
      <MemoryRouter>
        <BlogFeatured {...props} />
      </MemoryRouter>
    );
  };

  beforeEach(() => {
    props = {
      post: {
        content: {
          html: "<p>lorem ipsum dolor sit amet</p>"
        },
        date: "2020-01-01",
        id: "test-id",
        imageTemp: "https://via.placeholder.com/150",
        slug: "test-slug",
        sticky: false,
        title: "Test Title"
      }
    };

    createWrapper();
  });

  test("should render our BlogFeatured", () => {
    expect(component.baseElement).toMatchSnapshot();

    const elDate = component.getByText("12/31/2019"); // props.post.date
    const elHeading = component.getByText(props.post.title);
    const elImages = component.getAllByAltText(props.post.title ?? "_no_alt_");

    expect(elDate).toBeInTheDocument();
    expect(elHeading).toBeInTheDocument();
    expect(elImages[0]).toBeInTheDocument();
  });
});
