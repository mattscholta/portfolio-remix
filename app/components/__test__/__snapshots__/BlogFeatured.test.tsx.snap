// Vitest Snapshot v1

exports[`BlogFeatured Component > should render our BlogFeatured 1`] = `
<body>
  <div>
    <a
      class="blog-preview relative flex flex-col"
      href="/blog/test-slug"
    >
      <img
        alt="Test Title"
        height="auto"
        loading="lazy"
        src="https://via.placeholder.com/150"
        width="auto"
      />
      <h3
        class="mt-4 mb-2 text-2xl"
      >
        Test Title
      </h3>
      <div
        class="font-font-monospace text-sm"
      >
        12/31/2019
      </div>
    </a>
  </div>
</body>
`;
